### PR TITLE
Wildcard hostnames

### PR DESCRIPTION
**Pull request by ZackButcher**
One final PR, this time based off of release-0.8 branch rather than master since the bot merged changes.

Supersedes https://github.com/istio/istio/pull/5186, https://github.com/istio/istio/pull/5361

### CODE REVIEW

**Comment by costinm** [comment]
Can you resolve conflicts and rebase ? Tests should be better now.

File: pilot/pkg/serviceregistry/external/conversion.go [hunk 1]
```diff
@@ -50,27 +51,21 @@ func convertServices(serviceEntry *networking.ServiceEntry) []*model.Service {
...
+			host = strings.Replace(host, "/", "_", -1) // make hostname easy to parse for metrics
 		}
+
+		out = append(out, &model.Service{
+			MeshExternal: true,
+			Hostname:     model.Hostname(host),
+			Address:      address,
+			Ports:        svcPorts,
+			Resolution:   resolution,
+		})
```
Comment (rshriram) [inline pilot/pkg/serviceregistry/external/conversion.go]: this also needs to be reverted..

**Comment by frankbu** [comment]
Force merging to get past ongoing build problems.

**Pull request merged**